### PR TITLE
#18332: Fix BN hang for FPU Kernel

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
### Ticket
#18332

### Problem description
Batch norm kernel hangs for larger shapes. This was because the buffer size is set to 2. The kernel reserves `freq` number of tiles. So when this gets more than 2, there is an hang as it tries to reserve more than the size. Hence unpack and math threads got hung at the `copy_tile` call while the pack thread was stuck at `cb_reserve_back`.

### What's changed

- Reduced the number of `freq` loops in the kernel to ensure that as each tile gets reserved, they also get popped before the next tile gets reserved
- Removed `cb_num` buffer as I have rearranged the computation

### Checklist
- [x] [All post-commit tests](https://github.com/tenstorrent/tt-metal/actions/runs/13679949413)
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/runs/13695259508)
- [x] Full [new models](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) -  [Link to test](https://github.com/tenstorrent/tt-metal/actions/runs/13703995926) - as in main
- [x] [(Single-card) Demo tests](https://github.com/tenstorrent/tt-metal/actions/runs/13695278298) - as in main
- [x] [(Single-card) Device perf regressions](https://github.com/tenstorrent/tt-metal/actions/runs/13679957934) 
- [x] [Single-card Model perf tests](https://github.com/tenstorrent/tt-metal/actions/runs/13679960400)

### CODE REVIEW

**Review by umadevimcw — APPROVED** [review]

**Review by bbradelTT — APPROVED** [review]

**Review by mcw-anasuya — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/batch_norm_kernel.cpp [hunk 1]
```diff
@@ -65,32 +45,40 @@ ALWI void batchnorm_bcast_tiles(
...
+        sub_tiles(cb_other, cb_bcast, 0, 0, 0);
+
+        // (input - batch_mean)/(sqrt(batch_var + eps)) = result
+        binary_dest_reuse_tiles_init<EltwiseBinaryType::ELWMUL, EltwiseBinaryReuseDestType::DEST_TO_SRCA>(cb_den);
+        binary_dest_reuse_tiles<EltwiseBinaryType::ELWMUL, EltwiseBinaryReuseDestType::DEST_TO_SRCA>(cb_den, 0, 0);
         tile_regs_commit();
 
         tile_regs_wait();
-        pack_tile_with_dt(dst0, cb_affine_or_out);
+        pack_tile_with_dt(0, cb_affine_or_out);
```
Comment (tt-aho) [inline ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/batch_norm_kernel.cpp:74]: Minor: Why was this change needed?

**Review by tt-aho — APPROVED** [review]

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/13695246882 - passed

**Comment by VirdhatchaniKN** [comment]
Screenshot 2025-03-07 at 7 39 26 AM
